ordered_product/tests/test_ordered_product_page.py

```diff
@@ -0,0 +1,188 @@
+        assert response.status_code == 200
+        template_names = set(template.origin.template_name for template in response.templates)
+        assert 'ordered_product/orders.html' in template_names
+        client_orders_by_the_web_view = list(response.context['orders'])
+        client_orders_by_objects_filter = list(OrderedProduct.filter_user_name(saved_client0))
+        assert client_orders_by_the_web_view == client_orders_by_objects_filter
+        assert response.context['message'] == ""
+
+    @pytest.mark.django_db()
+    def test_change_quantity(self, client, valid_quantity, saved_client0, ordered_product0, client0_login):
+        assert ordered_product0.user_name == saved_client0
+        ordered_product0.save_ordered_product()
+        quantity_before_change = OrderedProduct.objects.get(id=ordered_product0.id).quantity
+        response = client.get('/ordered_product/orders/'+str(ordered_product0.id), {'quantity': valid_quantity})
+        quantity_after_change = OrderedProduct.objects.get(id=ordered_product0.id).quantity
+        assert quantity_after_change == quantity_before_change+int(valid_quantity)
+        assert response.status_code == 302
+        template_names = set(template.origin.template_name for template in response.templates)
+        assert template_names == set()
+
+    @pytest.mark.django_db()
+    def test_delete_order(self, client, saved_client0, ordered_product0, client0_login):
+        assert ordered_product0.user_name == saved_client0
+        ordered_product0.save_ordered_product()
+        assert ordered_product0 in OrderedProduct.objects.filter(user_name=saved_client0)
+        response = client.get('/ordered_product/orders/'+str(ordered_product0.id)+',delete')
+        assert ordered_product0 not in OrderedProduct.objects.filter(user_name=saved_client0)
+        assert response.status_code == 302
+        template_names = set(template.origin.template_name for template in response.templates)
+        assert template_names == set()
+
+    @pytest.mark.django_db()
+    def test_display_new_order_page(self, client, client0_login):
+        response = client.get('/ordered_product/new_order/')
+        assert response.status_code == 200
+        template_names = set(template.origin.template_name for template in response.templates)
+        assert 'ordered_product/new_order.html' in template_names
+
+    @pytest.mark.django_db()
+    def test_valid_search_product(self, client, saved_supplier_product0, client0_login):
+        VALID_PRODUCT_NAME = saved_supplier_product0.qr_code.product_name
+        product = Product.objects.get(product_name=VALID_PRODUCT_NAME)
+        supplier_products_by_objects_filter = SupplierProduct.objects.filter(qr_code=product)
+        assert saved_supplier_product0 in supplier_products_by_objects_filter
+        response = client.get('/ordered_product/new_order/search_product', {'product_name': VALID_PRODUCT_NAME})
+        assert response.status_code == 200
```

Comment:
If we already have a separate test for checking the 200 status code, there is no need ro include it again in other tests

Reply:
I don't have test to the exactly same URL (/ordered_product/new_order/search_product)

Reply:
maybe we should have